### PR TITLE
[service] Introduce ServiceManager and KvQueryClient

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

Part of https://cwiki.apache.org/confluence/display/PAIMON/PIP-10%3A+Introduce+Paimon+QueryService

1. ServiceManager to manage service addresses in FileSystem.
2. KvQueryClient to connect query servers and query.

### CODE REVIEW

File: paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.ConnectException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/** /** A class for the Client to get values from Servers. */
```
Comment (yuzelin) [inline paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java:39]: Clean comment.

File: paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        }
+    }
+
+    private CompletableFuture<KvResponse> getResponse(
+            final KvRequest request, final boolean forceUpdate) {
+        InetSocketAddress serverAddress =
+                queryLocation.getLocation(request.partition(), request.bucket(), forceUpdate);
+        if (serverAddress == null) {
+            return FutureUtils.completedExceptionally(
+                    new RuntimeException("Cannot find address for bucket: " + request.bucket()));
```
Comment (yuzelin) [inline paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java:106]: Also throw partition?

File: paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        return networkClient.sendRequest(serverAddress, request);
+    }
+
+    public void shutdown() {
+        try {
+            networkClient.shutdown().get(10L, TimeUnit.SECONDS);
+            LOG.info("{} was shutdown successfully.", networkClient.getClientName());
+        } catch (Exception e) {
+            LOG.warn("{} shutdown failed: {}", networkClient.getClientName(), e);
```
Comment (yuzelin) [inline paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java:116]:
```
 private void recordEvent2Args(Level level, Marker marker, String msg, Object arg1, Object arg2) {
        if (arg2 instanceof Throwable) {
            recordEvent(level, marker, msg, new Object[] { arg1 }, (Throwable) arg2);
        } else {
            recordEvent(level, marker, msg, new Object[] { arg1, arg2 }, null);
        }
    }
```
The placeholder for exception is unuseful.

File: paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java (hunk 2 above)
Comment (JingsongLi) [reply paimon-service/paimon-service-client/src/main/java/org/apache/paimon/service/client/KvQueryClient.java:106]: BinaryRow can not be show.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**